SPONSORS.list

```diff
@@ -163,6 +163,12 @@ SGI - http://www.sgi.com/
 	SGI has provided hardware donations for Squid developers.
 
 @Squid-2.3:
+National Laboratory for Applied Network Research
+
+	NLANR funded the early development of Squid from 1996-2005
+	with features for integration with the IRCache network
+	measurement project and High Performance Networking.
```

Comment:
AFAIK, being an informal organization, NLANR could not really "fund" anything. Also, its contributions go way beyond the two features mentioned above. Finally, a phrase "from x-y" does not say what we want to say (and I would recommend being less specific regarding the last NLANR contribution year -- I am not sure it was 2005).

I do not insist on any specific wording, but I think we have two reasonable options here:

1. Edit this bullet to say something vague like "NLANR orchestrated Squid development from 1996 up to early 2000s. In collaboration with other web pioneers, NLANR laid the foundation for the Squid Project and introduced numerous Squid features."

2. Merge this entire bullet into the NSF bullet below by adding a new sentence to its existing "The NSF was..." paragraph. Something along these lines may work: "These grants enabled the National Laboratory for Applied Network Research (NLANR) team, in collaboration with other web pioneers, to lay the foundation for the Squid Project and introduce numerous Squid features."

Informally, the first option may be better because it features NLANR more prominently, and NLANR deserves that recognition. On a formal level, the second option may be slightly better because it lists NSF (i.e. a real organization with money) as the "sponsor" (and this is a SPONSORS file). Personally, I am OK with either option. Your call.

I am not going to block this PR if others insist on the "NLANR funded" text, but I think it is important that we fix this.

Reply:
FYI, I found a few grant budget documents clearly covering 1996-2005 funding NLANR network measurement projects including IRcache and early VBNC measurements collection on Internet2. Those documents do not match the NSF grants, and I was not able to find info on the money source. Thus the level of detail I used counter to your (1) and not merging per your (2).

In 2006 NLANR appears to have dissolved or handed those projects to other organizations. I am unaware of records about which nor whether they kept up Squid related stuff. If you or Duane know any of that they should probably get their own `SPONSORS.list` mentions if not there already.

The rest of the support which could be attributed to NLANR is already covered by the NSF entry as far as I can tell. The grants and budget timing looks like NSF started projects with some seed funding then NLANR later continued the ones that produced useful results.

As for the features I listed - yes it is incomplete. I am only able to list the ones **I** know were created for IRCache and VBNC Internet2 measurements covered by the grants. Additions are welcome if you know of specifics.

Reply:
If you found documents showing NLANR funding Squid infrastructure or development, please point me to those documents. NSF grants funding NLANR-affiliated projects do not count (in this context) because that would be NSF being the sponsor, not NLANR.

Reply:
The ones CNS-9720674, CNS-9807479, ACI-0129681 and ACI-0129677. Funding not for specific projects, but for continuance and ongoing operations of NLANR projects started by the NSF.

Reply:
In all those four cases, NSF is doing the funding, not NLANR. Again, to the best of my knowledge, NLANR could not fund something because it was an _informal_ organization, a group of internet researches/operators/etc. representing different organizations and collaborating on various projects (including but not limited to Squid); NLANR had no bank accounts, wrote no checks, employed nobody, and owned no property. If you look at the actual recipients of those NSF funds, they are not NLANR either -- the recipients are "real" organizations like University of California in San Diego (UCSD that is already mentioned in another bullet) and University of Illinois at Urbana-Champaign.

My original "do not say `NLANR funded`" change request and its specific suggestions/recommendations are still valid.

P.S. I do not know whether any of the four grants you found funded Squid activities. Some of them funded people and groups that I knew and that did participate in Squid-related projects, but that does not necessarily mean that those grants were used (in part) for Squid things. It is common for the same researcher or team to receive NSF funding from different grants, and each grant will have its own spending targets/restrictions. Thus, I do _not_ recommend listing those specific grant numbers officially in this file unless you know for sure that they funded Squid.

Reply:
I have removed the `M-cleared-for-merge` label to make sure my change request update (added 1 hour after the label) is not ignored. I will not block this PR from going in despite my objections, but since this is about money, I consider this issue important enough to get an explicit Francesco opinion. I am requesting his review.

Reply:
Done

Reply:
I consider this change request addressed.